components/automate-ui/src/app/entities/projects/project.effects.ts

```diff
@@ -6,7 +6,7 @@ import { interval as observableInterval, of as observableOf, Observable } from '
 import { catchError, mergeMap, map, filter, switchMap, withLatestFrom } from 'rxjs/operators';
 
 import { NgrxStateAtom } from 'app/ngrx.reducers';
-import { HttpStatus } from 'app/types/types';
+import { HttpStatus, GrpcErrorResponse, GrpcStatus } from 'app/types/types';
```

Comment:
feels like we're leaking some backend concerns with these GRPC responses. HTTP does have a [412 status code](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/412) for failed precondition, but I noticed the API returns a 400 (using the changes from your PR description)
```
< HTTP/2 400 
< server: nginx
< date: Mon, 09 Dec 2019 19:26:13 GMT
< content-type: application/json
< trailer: Grpc-Trailer-Content-Type
< x-xss-protection: 1; mode=block
< strict-transport-security: max-age=63072000; includeSubDomains
< x-content-type-options: nosniff
< 
{ [119 bytes data]

* Connection #0 to host localhost left intact
{
  "error": "project still has rules \"project1\"",
  "code": 9,
  "message": "project still has rules \"project1\"",
  "details": []
}
```

I'm thinking we change this UI code to look for code 412, then add a requirement to the backend card that it must return that HTTP status code

Reply:
See below. 🤷‍♂